src/hotspot/share/opto/callnode.hpp

```diff
@@ -1062,6 +1062,10 @@ class AllocateNode : public CallNode {
   bool is_allocation_MemBar_redundant() { return _is_allocation_MemBar_redundant; }
 
   Node* make_ideal_mark(PhaseGVN *phase, Node* obj, Node* control, Node* mem);
+
+#ifndef PRODUCT
+  virtual void dump_spec(outputStream* st) const;
+#endif
```

Comment:
For single line not product `defs`, you can use:
```suggestion
  NOT_PRODUCT(virtual void dump_spec(outputStream* st) const;)
```

Reply:
Done. It is shorter, and I don't see a good reason not to use it.